### DISCUSSION TITLE
How to use EnginesGroups with FasterList properly

### DISCUSSION THREAD

**Discussion by kcze** [sebas77/Svelto.ECS#113] [Q&A] [answered]
Why `EnginesGroup<T>` don't allow `IEngine` in the generic parameter? Because of that it's impossible to create a `FasterList` that would allow EnginesGroups with both `IStepEngine` and `IStepEngine<>`.

**Comment by sebas77** [comment, marked as answer]
To answer properly this question first you need to remember that Svelto.ECS doesn't have any ticking system like other ECS framework may do. It's 100% on the final user to tick the "engines" and to submit the entities. the API is designed in such a way that you have complete control on when these things happens during a frame. 

all that said EnginesGroup is then just a tool the framework provides to facilitate the task to organise group of engines that must tick sequentially or in order. This tool is totally optional, you don't need to use it, you can as well write your own way to tick IEngines.

however if you use it, then you need to implement IStepEngines because the groups expect the "Step" method to be implemented as a contract.

Eventually you still need to tick the whole group manually. You won't need to tick the single engines inside it, but you still need to tick the group itself.
